logstash-core/src/main/java/org/logstash/RubyUtil.java

```diff
@@ -47,13 +47,10 @@
     static {
         RUBY = Ruby.getGlobalRuntime();
         LOGSTASH_MODULE = RUBY.getOrCreateModule("LogStash");
-        RUBY_TIMESTAMP_CLASS = setupLogstashClass("Timestamp", new ObjectAllocator() {
-            @Override
-            public JrubyTimestampExtLibrary.RubyTimestamp allocate(final Ruby runtime,
-                final RubyClass rubyClass) {
-                return new JrubyTimestampExtLibrary.RubyTimestamp(runtime, rubyClass);
-            }
-        }, JrubyTimestampExtLibrary.RubyTimestamp.class);
+        RUBY_TIMESTAMP_CLASS = setupLogstashClass(
+            "Timestamp",
+            JrubyTimestampExtLibrary.RubyTimestamp::new, JrubyTimestampExtLibrary.RubyTimestamp.class
```

Comment:
Can you help me understand how you can replace an 'ObjectAllocator' with a `JrubyTimestampExtLibrary.RubyTimestamp::new` ?

Reply:
@jakelandis sure, the API requires `ObjectAllocator` which is a functional interface.
=> you can use a lamba, that has the content of its only method `public IRubyObject allocate(Ruby runtime, RubyClass klazz)` as its body => since the body is only a single method call to `JrubyTimestampExtLibrary.RubyTimestamp::new` the lambda can be replaced by a reference to that method call.
The only reason I used an `ObjectAllocator` instead of the reference before, was that we had multiple constructors and on some older Java 8 the call is a lot slower in that case (I think there were cases were if there is more than one override for `rubyTimestampExtLibrary.RubyTimestamp::new` the lamdba wasn't cached but now it should compile to an `invokestatic` since we have only a single constructor/override).

Reply:
OK, i see now. That took me way longer then I like to admit to understand that (even with the explanation). 

Thanks for the explains.

Reply:
np :)